geth/node/status_node.go

```diff
@@ -268,44 +263,28 @@ func (n *StatusNode) Config() (*params.NodeConfig, error) {
 	return n.config, nil
 }
 
-// LightEthereumService exposes reference to LES service running on top of the node
-func (n *StatusNode) LightEthereumService() (*les.LightEthereum, error) {
-	n.mu.RLock()
-	defer n.mu.RUnlock()
-
+// gethService is a wrapper for gethNode.Service which retrieves a currently
+// running service registered of a specific type.
+func (n *StatusNode) gethService(serviceInstance interface{}, serviceName string) error {
 	if err := n.isAvailable(); err != nil {
-		return nil, err
-	}
-	if n.lesService == nil {
-		if err := n.node.Service(&n.lesService); err != nil {
-			n.log.Warn("Cannot obtain LES service", "error", err)
-			return nil, ErrInvalidLightEthereumService
-		}
+		return err
 	}
-	if n.lesService == nil {
-		return nil, ErrInvalidLightEthereumService
+	if err := n.gethNode.Service(serviceInstance); err != nil || serviceInstance == nil {
+		n.log.Warn("Cannot obtain ", serviceName, " service", "error", err)
+		return ErrInvalidService
 	}
-	return n.lesService, nil
+
+	return nil
 }
 
-// WhisperService exposes reference to Whisper service running on top of the node
-func (n *StatusNode) WhisperService() (*whisper.Whisper, error) {
-	n.mu.RLock()
-	defer n.mu.RUnlock()
+// LightEthereumService exposes reference to LES service running on top of the node
+func (n *StatusNode) LightEthereumService() (l *les.LightEthereum, err error) {
+	return l, n.gethService(&l, "LES")
```

Comment:
LES is only for logging, right? you can also use reflect.TypeOf

Reply:
It's constant here, so I would avoid using reflection. 😈 😄

Reply:
not here, the reflection will be used in gethService method

Reply:
Yep, was thinking about implementing something similar to how it's done [on geth](https://github.com/ethereum/go-ethereum/blob/master/node/node.go#L596). 
However, I ended up avoiding it in favor of this simpler and easy to read approach.